src/wsrep_provider_v26.cpp

```diff
@@ -71,7 +71,12 @@ namespace
         case WSREP_NOT_ALLOWED:
             return wsrep::provider::error_not_allowed;
         }
-        return wsrep::provider::error_unknown;
+
+        // preserve negative error code
+        if (status < 0)
```

Comment:
Given that enum types are not guaranteed to be signed integers, I would change this to return `wsrep::provider::error_unknown` and print a warning message which contains the raw status value.